### PR TITLE
Add code coverage CI step

### LINKED ISSUE #176: Add code coverage badge
We should measure code coverage by CI and add a badge to the repo, goal being to increase have RON fully covered by tests.

### PR DESCRIPTION
**Pull request by MomoLangenstein**
Fixes #176

~~* [ ] I've included my change in `CHANGELOG.md`~~

### CODE REVIEW

**Comment by MomoLangenstein** [comment]
@kvark @torkleyy I'm hoping that just adding the CI step and badge should just work. Ideally, we wouldn't need to reinstall `grcov` on every CI run, but the [actions-rs/install](https://github.com/actions-rs/install) action that could solve this still has security issues - potentially this might become an incremental improvement later on. I've added `GRCOV_EXCL_LINE`,  `GRCOV_EXCL_START`, and `GRCOV_EXCL_STOP` markers so we can exclude sections from coverage analysis.

**Comment by MomoLangenstein** [comment]
If we don't find any issues with this PR, I'll merge it by February 28th.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -58,3 +58,35 @@ jobs:
...
+        with:
+          toolchain: nightly
+          profile: minimal
+          components: llvm-tools-preview
+          override: true
+      - run: cargo install grcov --force --locked
+      - run: |
+          cargo test --all-features --all-targets
+          cargo run --example decode_file
+          cargo run --example decode
```
Comment (torkleyy) [inline .github/workflows/ci.yaml:77]: I don't think examples should contribute to coverage.

**Review by torkleyy — APPROVED** [review]
Looks good otherwise!

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (MomoLangenstein) [reply .github/workflows/ci.yaml]: Ok, I removed the examples from the coverage runs and output - so they just won't be recorded at all and will also not drag down the coverage

**Review by torkleyy — APPROVED** [review]
:shipit:

**Pull request merged**